### PR TITLE
Anand_Add validation for required review link and implement link checking

### PR DESCRIPTION
**Pull request by anandsesha**
# Description
Description

## Related PRS (if any):
This frontend PR is related to the dev backend
…

## Main changes explained:
- Updated the input field to be 'required' and show an error message when an invalid or empty URL is entered during review submission.
- Modified `handleLink` to set the appropriate error message depending on whether the URL is empty or invalid, clearing the error when a valid URL is entered.
- Implemented logic to clear linkError when the review modal is closed to prevent error messages from persisting on re-opening.
…

## How to test:
1. Check into current branch
2. Do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. Log as any user with task assigned
5. Go to dashboard→ Task Tab→ Submit For Review.
6. Click 'Submit for Review' and the Change Review Status model pops up
- The model should display → Please add link to related work: (required)
- Check if you are able to Submit review without entering any link. Appropriate error message should be shown.
- Test using other invalid links/shortcuts/without https:// to see if it accepts such links. 
- When an error message is shown, click 'Cancel' and re-open the Change Review Status modal by clicking on Submit for Review. The earlier error should not be persisted when the popup is reopened. 

## Screenshots or videos of changes:
Screenshot (858)
Screenshot (862)
https://www.loom.com/share/f627bb2b70c247cab222c2fff4d68365?sid=e5251dee-98cd-4dec-a1f9-60a81f1f823e

## Note:
N/A

### CODE REVIEW

**Comment by alla0810** [comment]
Hi, [anandsesha](https://github.com/anandsesha).
Can I ask a question?
I've tried to find `Submit for Review` from the Task tab, but I don't know how to add this `Submit for Review` under a user's Task.
Can you please explain how I can add `Submit for Review` to test this PR?

**Comment by anandsesha** [comment]
> Hi, [anandsesha](https://github.com/anandsesha). Can I ask a question? I've tried to find `Submit for Review` from the Task tab, but I don't know how to add this `Submit for Review` under a user's Task. Can you please explain how I can add `Submit for Review` to test this PR?

Hi @alla0810 you can follow the below video link which explains how to add a task and assign yourself to it. Once you do, it will show up on the Dashboard under the Task tab.
https://www.loom.com/share/8bc672c323b64955aac4bce46c3794aa?sid=b05f5604-87cb-4ca1-80b8-49bd823893ee

**Review by alla0810 — DISMISSED** [review]
Hi, [anandsesha](https://github.com/anandsesha).
To tell the scope of this PR, looks good working.

When logged in `DevAdmin`, I could assign a task to `KyoTest Admin`, and when logged in as `KyoTest Admin`, I could see `Submit for Review` button.
As expected by this PR, I had to add an appropriate link related to this task.
I've tested 3 invalid cases, one empty link and two incomplete links, and checked these linked were successfully declined.

4-good
5-good
6-good

When I input an appropriate link, the `Submit for Review` was approved and the button turn into `Ready for Review` state, which is a normal state transition.

7-good

Apart from the requirement of this PR, this branch seems very unstable, sometimes made my all of the chrome browsers `Not Responding`, or sometimes I had to reboot my computer.
I've also noticed some errors repeatedly coming up including `GET` method error.
Please see the screenshots of DevTools, network side error messages, bash windows error messages, and console log file attached.





[console log.log](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/files/15139023/console.log.log)

**Review by Ramaruva — DISMISSED** [review]
I have verified changes in my local url checking is working as expected

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/64008831/78db930f-29a6-4e31-90ed-d9c60161e011

**Review by peter6866 — DISMISSED** [review]
Changes work as expected with invalid links.
image

**Comment by nishithaShetty** [comment]
Hi Anand, Great work.
It works as expected. I see the error message when we submit without a link, and when we click the submit review button again, the error message does not reappear. I also tried submitting the URL without https://, and an error is correctly thrown.
PR#2185_1
PR#2185_2

**Review by JatinAgrawal94 — DISMISSED** [review]
Feature works as expected.
Great work!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/47699187/cb3aff10-f990-4a3a-8dd3-a8ccec9af2ab

**Comment by JatinAgrawal94** [comment]
> Feature works as expected. Great work!
> 
>  PR.2185.mp4

**Review by CloudyZ524 — APPROVED** [review]
Hi Anand, 
The function of adding valid URL for "Submit for Review" works as expected. 
I've been able to assign a new task to a user and tested from the user's task tab. Also, it is impossible to submit an invalid URL. Well done!

Screenshot 2024-06-07 at 15 49 51

File: src/components/TeamMemberTasks/ReviewButton.jsx [hunk 1]
```diff
@@ -186,16 +198,23 @@ const ReviewButton = ({
...
+          required
           value={link}
           onChange={handleLink}
           />
+          {linkError && <div className="text-danger">{linkError}</div>}
         </ModalBody>
         <ModalFooter>
           <Button
             onClick={(e) => {
+              e.preventDefault();
```
Comment (TXMO-dev) [inline src/components/TeamMemberTasks/ReviewButton.jsx:213]: If this can be put in a seperate class for reusability and maintainability please.

**Review by TXMO-dev — CHANGES_REQUESTED** [review]

**Comment by akshit0211** [comment]
Hi Anand, the changes are working as expected. Good work!

err1
err2
err3

**Review by akshit0211 — DISMISSED** [review]

**Review by TXMO-dev — APPROVED** [review]
URL validation check works perfectly fine and meets all requirements. Great Job. Approved!

**Review by suaniii — DISMISSED** [review]
The functionality works as expected: an error message appears when submitting without an appropriate link, and it does not reappear upon clicking the submit review button again. I tested an empty link and incomplete links, and confirmed that these links were not accepted.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113208952/910a3689-76f8-479b-b9e5-ee005f677be5

**Review by nyelmela — DISMISSED** [review]
I have tested the functionality in this PR by running the application locally, clearing site data, and performing the steps to submit a task for review. The "Change Review Status" modal correctly prompts for a link to related work and shows appropriate error messages for invalid or missing links. Errors do not persist when reopening the modal. Everything is functioning as expected. 
2185 Anand_required_review_link_on_task_submission

**Review by Vigneshwar-Muriki — DISMISSED** [review]
Tested this PR using admin account. Firstly, I have created a task for the admin and then refreshed it so that I could see the task in the tasks section. Then I clicked submit review button and a dialog box appears to enter a valid URL. I have tried all possible cases and it works well for all scenarios. Also, I clicked cancel too see if the error persists after clicking submit review button button and it hasn't.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/131609223/3532063e-1705-4dbf-bc12-856b111f48c9

**Review by Shadhrush5 — DISMISSED** [review]
Hi, I have tested PR 2185, the changes on the current branch by following the steps provided: performed npm install and ran the PR locally, cleared site data/cache, logged in as an user with a task assigned, navigated to the dashboard -> tasks ->Submit review.  The pop up appears as expected and works as expected. The link checks are working fine as well.
Screenshot 2024-07-11 at 12 45 05 PM
Screenshot 2024-07-11 at 12 45 20 PM
Screenshot 2024-07-11 at 12 46 04 PM
Screenshot 2024-07-11 at 12 46 17 PM

I have approved the same. Great work.

**Review by AnujVakil — DISMISSED** [review]
I have tested this PR and It is working correctly as expected.
PR #2185_1
PR #2185_2
PR #2185_3

**Comment by KurtisIvey** [comment]
I noticed there was a merge conflict on this pull request. Would you be able to clear it up please when you get a chance? Looks to be a high quality pull request as you've got all approvals, so we'd like to get this moved to final review and merged 😁

**Review by nikhilpittala16 — APPROVED** [review]
I have tested the PR and everything works as expected.
Screenshot (105)
Screenshot (106)
Screenshot (107)
Screenshot (108)

**Review by Shadhrush5 — CHANGES_REQUESTED** [review]
Hi, I tried reviewing the PR #2185. I ran into some trouble trying to build the front end on the current branch. I have attached a screenshot of the same.

Screenshot 2024-07-26 at 11 32 56 AM

**Comment by souravwalke** [comment]
Tried reviewing the PR but the local deployment failed with the following error in the console.

Sc

<img width=

Screenshot 2024-07-26 at 12 54 04 PM

**Review by varunreddy03 — CHANGES_REQUESTED** [review]
Hi I have tried to test this PR but the frontend of the PR has Failed to compile and it is giving  an error. I have attached screenshot for your reference.

Screenshot 2024-08-10 200158

**Comment by Lemonnycodes** [comment]
Hi Anand,

I verified the PR and it works as mentioned in the description 

1.  Attempted to submit without a link, and the appropriate error message was displayed.
2. Tested various invalid links (e.g., without https://, incomplete URLs, non-URLs) and confirmed that they were correctly rejected.
3.  Confirmed that after closing and reopening the modal, previous errors were not persisted, and the modal worked as expected.
Screenshot 2024-08-13 235740
Screenshot 2024-08-13 235802
Screenshot 2024-08-14 000401

 Encountered an issue where the project failed to compile due to a file naming mismatch: RoleInfoModal.jsx was expected, but the file was named 'roleInfoModal.jsx' on disk. Resolved the issue by renaming the file to 'RoleInfoModal.jsx' to match the expected casing. After addressing the file naming issue, the PR works as intended.
 
error

**Review by adityasure9 — APPROVED** [review]
This PR effectively implements the required link validation and handles all edge cases as expected. 

image

**Review by Lemonnycodes — APPROVED** [review]

**Review by oliviahyw — CHANGES_REQUESTED** [review]
Hi @anandsesha I have tested this PR and I got this error. Please check the screenshot attached.
PR2185

**Comment by FayeLV** [comment]
Hi,
Great job, all tests have passed, but I found a small issue when I first opened the project and a compilation error occurred. I changed **roleInfoModal**.jsx to **RoleInfoModal**.jsx, and the project ran successfully. Other team members have also reported this issue, and the error still persists.
**Without entering a link:** The system correctly shows an error message stating that a link is required.
**Invalid links:** The system rejects links that are missing a protocol, incomplete, lacking a top-level domain, or containing spaces, special characters, or an incorrect protocol. Appropriate error messages are displayed.
**Valid links:** The system successfully accepts and submits links with https://  protocols.
**Error message persistence check:** After closing the modal, the error message does not persist when reopening it, and the modal resets correctly.
Snipaste_2024-09-06_12-14-06

Snipaste_2024-09-06_12-09-31
Snipaste_2024-09-06_12-09-42
Snipaste_2024-09-06_12-09-56
Snipaste_2024-09-06_12-10-07
Snipaste_2024-09-06_12-10-19
Snipaste_2024-09-06_12-10-38
Snipaste_2024-09-06_12-11-22

**Pull request closed without merging**